core/controllers/base.py

```diff
@@ -230,6 +233,11 @@ def dispatch(self):
 
         if self.payload is not None and self.REQUIRE_PAYLOAD_CSRF_CHECK:
             try:
+                if 'signup' in self.request.uri and not self.user_id:
+                    raise self.LoggedOutInNewTabException(
```

Comment:
Just raise an exception without text instead; keep the descriptive text in the frontend. Let the frontend deal with the error based on its being a 401. I also don't think you actually need a separate error class here, but please add a comment in the Python code to explain in detail what's going on.

Reply:
Will it be fine to use `UnauthorizedUserException` class instead of using the new one?

Reply:
Yes I think that's OK (and drop the new class).

Reply:
If we just check the status code being 401, it will also be valid for the case when the csrf token has expired (401 is raised in that case too) and that will result in continue registration modal being shown in that case too. I feel it is fine to show the modal in both the cases. What do you think?